File: src/ruby/NinjaSurvivalRate/features/step_definitions/killer_steps.rb
require 'rspec/expectations' 
require 'cucumber/formatter/unicode'

@@killed=0
@@remaining=3

Before do
  @@killed=@@killed+1 
end

After do |scenario|
  @@remaining=@@remaining-1
end

When /^this scenario is executed$/ do
  # do absolutely nothing
end

Then /^[A-z ]* should kill one ninja$/ do
  @@killed.should == 1 
end

Then /^[A-z ]* should kill (\d+) ninjas$/ do |expected|
  expected.to_i.should == @@killed 
end

Then /^[A-z ]* should expect (\d+) ninjas$/ do |expected|
  expected.to_i.should == @@remaining
end  

File: src/dotnet/SpecFlow/NinjaSurvivalRate/KillerSteps.cs
using NUnit.Framework;
using TechTalk.SpecFlow;

namespace NinjaSurvivalRate
{
    [Binding]
    public class KillerSteps
    {
        private static int killed;
        private static int remaining = 3;

        [BeforeScenario("First_blood","Second_blood")]
        public void KillNinjaBeforeEachScenario()
        {
            killed++;
        }

        [AfterScenario("First_blood", "Second_blood")]
        public void ExpectLessNinjasAfterScenario()
        {
            remaining--;
        }

        [When(@"^this scenario is executed$")]
        public void ThisScenarioIsExecuted()
        {
            // do absolutely nothing
        }

        [Then("^Chuck Norris should expect ([0-9]+)* ninjas$")]
        public void ChuckNorrisShouldExpectNinjas(int expected)
        {
            Assert.AreEqual(expected, remaining);
        }

        [Then("^Chuck Norris will have killed one ninja$")]
        public void ChuckNorrisShouldKillOneNinja()
        {
            HeShouldKillNinjas(1);
        }

        [Then("^he will have killed ([0-9]+)* ninjas$")]
        public void HeShouldKillNinjas(int expected)
        {
            Assert.AreEqual(expected, killed);
        }

    }
}